app/src/main/java/org/mozilla/fenix/ext/BookmarkNode.kt

```diff
@@ -75,49 +51,69 @@ suspend fun BookmarkNode?.withOptionalDesktopFolders(
 
 private suspend fun virtualDesktopFolder(context: Context?): BookmarkNode? {
     val rootNode = context?.bookmarkStorage()?.getTree(BookmarkRoot.Root.id, false) ?: return null
-    return BookmarkNode(
-        type = rootNode.type,
-        guid = rootNode.guid,
-        parentGuid = rootNode.parentGuid,
-        position = rootNode.position,
-        title = rootTitles[rootNode.title],
-        url = rootNode.url,
-        children = rootNode.children
-    )
+    return rootNode.copy(title = rootTitles[rootNode.title])
 }
 
 /**
  * Removes 'mobile' root (to avoid a cyclical bookmarks tree in the UI) and renames other roots to friendly titles.
  */
 private fun processDesktopRoots(roots: List<BookmarkNode>?): List<BookmarkNode>? {
-    if (roots == null) {
-        return null
-    }
+    if (roots == null) return null
 
     return roots.filter { rootTitles.containsKey(it.title) }.map {
-        BookmarkNode(
-            type = it.type,
-            guid = it.guid,
-            parentGuid = it.parentGuid,
-            position = it.position,
-            title = rootTitles[it.title],
-            url = it.url,
-            children = it.children
-        )
+        it.copy(title = rootTitles[it.title])
     }
 }
 
+/**
+ * Restructures roots to place desktop bookmark roots underneath the mobile root for selecting a folder
```

Comment:
This comment and function name are a bit confusing to me. It's not clear to me what's going on here. It looks like the main thing that we're doing here is updating the titles to be user friendly. Perhaps call that out in the function name and comment?